Update and simplify CodeGen tutorials (#351)
opea-project/docs: tutorial/CodeGen/deploy/gaudi.md

# Single node on-prem deployment on Gaudi AI Accelerator

> This section covers single-node on-prem deployment of the CodeGen example. It will show how to build an end-to-end CodeGen solution with the `Qwen2.5-Coder-32B-Instruct` model deployed on Intel® Gaudi® AI Accelerators. To quickly learn about OPEA and set up the required hardware and software, follow the instructions in the [Getting Started](../../../getting-started/README.md) section. 

Reviewer: ```suggestion
This section covers single-node on-prem deployment of the CodeGen example. It will show how to deploy an end-to-end CodeGen solution with the `Qwen2.5-Coder-32B-Instruct` model running on Intel® Gaudi® AI Accelerators. To quickly learn about OPEA and set up the required hardware and software, follow the instructions in the [Getting Started](../../../getting-started/README.md) section. 
```

## Overview

The CodeGen use case uses a single microservice called LLM with model serving done with vLLM or TGI.

> This solution is designed to demonstrate the use of the `Qwen2.5-Coder-7B-Instruct` model for code generation on Intel® Gaudi® AI Accelerators. The steps will involve setting up Docker containers, taking text input as the prompt, and generating code. Although multiple versions of the UI can be deployed, this tutorial will focus solely on the default version.

Reviewer 2: line 3, it is Qwen2.5-Coder-32B-Instruct`.
Here it is `Qwen2.5-Coder-7B-Instruct`. 
Are they two different cases?

Reviewer: I too noticed the 32B model mentioned further down when HuggingFace token mentioned.

Author: fixed. 7B for Xeon, 32B for Gaudi

## Prerequisites

> To run the UI on a web browser external to the host machine such as a laptop, the following port(s) need to be port forwarded when using SSH to log in to the host machine:

Reviewer 3: ```suggestion
To run the UI on a web browser external to the host machine such as a laptop, the following port(s) need to be forwarded when using SSH to log into the host machine:
```

- 7778: CodeGen megaservice port

This port is used for `BACKEND_SERVICE_ENDPOINT` defined in the `set_env.sh` for this example inside the `docker compose` folder. Specifically, for CodeGen, append the following to the ssh command: 
```bash
-L 7778:localhost:7778
```

Set up a workspace and clone the [GenAIExamples](https://github.com/opea-project/GenAIExamples) GitHub repo.
```bash
export WORKSPACE=<Path>
cd $WORKSPACE

> git clone https://github.com/opea-project/GenAIExamples.git # GenAIExamples

Reviewer 4: Comment is redundant. I'd remove ` # GenAIExamples`

```

**Optional** It is recommended to use a stable release version by setting `RELEASE_VERSION` to a **number only** (i.e. 1.0, 1.1, etc) and checkout that version using the tag. Otherwise, by default, the main branch with the latest updates will be used.
```bash

> export RELEASE_VERSION=<Release_Version> # Set desired release version - number only

Reviewer 4: For inline comments, leaving 2 spaces before `#` is preferred

    WARN[0000] The "http_proxy" variable is not set. Defaulting to a blank string.
    WARN[0000] The "https_proxy" variable is not set. Defaulting to a blank string.

Check if all the containers launched via `docker compose` are running i.e. each container's `STATUS` is `Up` and `Healthy`.

Run this command to see this info:
```bash
docker ps -a
```

Sample output: 
```bash
CONTAINER ID   IMAGE                                                   COMMAND                  CREATED              STATUS                        PORTS                                                                                      NAMES
c6fed95320ee   opea/codegen-gradio-ui:latest                           "python codegen_ui_g…"   About a minute ago   Up About a minute             0.0.0.0:5173->5173/tcp, [::]:5173->5173/tcp                                                codegen-gaudi-ui-server
092d76d64623   opea/embedding:latest                                   "sh -c 'python $( [ …"   About a minute ago   Up About a minute             0.0.0.0:6000->6000/tcp, [::]:6000->6000/tcp                                                tei-embedding-server

> fdce54b2c46f   opea/dataprep:latest                                    "sh -c 'python $( [ …"   About a minute ago   Up About a minute             0.0.0.0:6007->5000/tcp, [::]:6007->5000/tcp                                                dataprep-redis-server

Reviewer 3: ```suggestion
fdce54b2c46f   opea/dataprep:latest                                    "sh -c 'python $( [ …"   About a minute ago   Up About a minute (healthy)             0.0.0.0:6007->5000/tcp, [::]:6007->5000/tcp                                                dataprep-redis-server
```

```bash
curl http://${host_ip}:7778/v1/codegen \
  -H "Content-Type: application/json" \
  -d '{"agents_flag": "True", "index_name": "my_API_document", "messages": "Implement a high-level API for a TODO list application. The API takes as input an operation request and updates the TODO list in place. If the request is invalid, raise an exception."}'
  ```

## Launch UI
### Gradio UI
To access the frontend, open the following URL in a web browser: http://${host_ip}:5173. By default, the UI runs on port 5173 internally. A different host port can be used to access the frontend. Simply modify the port mapping in the `compose.yaml` file as shown below:
```yaml
  codegen-gaudi-ui-server:
    image: ${REGISTRY:-opea}/codegen-gradio-ui:${TAG:-latest}
    ...
    ports:
      - "5173:5173"
```

## Stop the Services

To stop and remove all the containers, use the commands below:

::::{tab-set}
:::{tab-item} vllm
:sync: vllm

```bash
cd $WORKSPACE/GenAIExamples/CodeGen/docker_compose/intel/cpu/xeon
docker compose --profile codegen-xeon-vllm down
```
:::
:::{tab-item} TGI
:sync: TGI

```bash
cd $WORKSPACE/GenAIExamples/CodeGen/docker_compose/intel/cpu/xeon

> docker compose --profile codegen-xeon-tgi down

Author: fixed
